### PR TITLE
Fix M428 width DELTA & SCARA

### PR DESCRIPTION
**Pull request by MagoKimbra**

### CODE REVIEW

**Comment by thinkyhead** [comment]
@MagoKimbra It seems as though we should replace `sync_plan_position` with `sync_plan_position_delta` everywhere that it's used. But there are some places in `DELTA` context where `sync_plan_position` is being called instead (and without `calculate_delta`). It seems like a good idea to re-examine those.

@jbrazio @esenapaj

**Comment by MagoKimbra** [comment]
I check it....
OK!

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -6005,7 +6005,11 @@ inline void gcode_M428() {
   }
 
   if (!err) {
-    sync_plan_position();
+    #if ENABLED(DELTA) && ENABLED(SCARA)
```
Comment (ghost) [inline Marlin/Marlin_main.cpp:6008]: @thinkyhead 
I'm sorry, I tried to understand your instruction and content of this merged PR,
but I stumbled on the first step.
Why isn't this `ENABLED(DELTA) || ENABLED(SCARA)` but `ENABLED(DELTA) && ENABLED(SCARA)`?
[This combination is forbidden in SanityCheck](https://github.com/MarlinFirmware/Marlin/blob/RCBugFix/Marlin/SanityCheck.h#L402).

Comment (MagoKimbra) [inline Marlin/Marlin_main.cpp:6008]: Yes you're right, my mistake

Comment (thinkyhead) [inline Marlin/Marlin_main.cpp:6008]: Patched!

**Comment by ghost** [comment]
@MagoKimbra
~~I also made a follow-up for this PR by way of trial. Please refer to this if you need.~~
~~https://github.com/esenapaj/Marlin/pull/137~~

# This asinine my PR is extremely danger.

# It breaks the function of endstops on delta machine.

# Please don't refer to this.

# I'm sorry.

**Pull request merged**